src/handlers/package_handler.js

```diff
@@ -843,31 +856,37 @@ async function getPackagesVersionTarball(req, res) {
   // the download to take place from their servers.
 
   // But right before, lets do a couple simple checks to make sure we are sending to a legit site.
-  //let tarballURL = new url((pack.content.meta.tarball_url ? pack.content.meta.tarball_url : ""));
-  let tarballURL = pack.content.meta.tarball_url;
-
-  if (tarballURL !== undefined && false) {
-    // todo the url feature doesn't work the way we would expect in production
-    tarballURL = url.parse(pack.content.meta.tarball_url);
-
-    if (
-      url.hostname != "codeload.github.com" ||
-      url.hostname != "api.github.com" ||
-      url.hostname != "github.com" ||
-      url.hostname != "raw.githubusercontent.com"
-    ) {
-      await common.handleError(req, res, {
-        ok: false,
-        short: "Server Error",
-        content: `Invalid Domain for Download Redirect: ${url.toString()}`,
-      });
-      return;
-    }
+  const tarballURL = pack.content.meta.tarball_url ?? "";
+  let hostname = "";
+
+  // Try to extract the hostname
+  try {
+    const url = new URL(tarballURL);
```

Comment:
We are importing the module as `url` then using `new URL` to initialize.

We should probably change the import to use `const URL = ...`

Reply:
In node.js >= 10.0.0, `URL` is a global with a spec-compliant implementation that does not need to be imported (here, the import could actually be removed). This is different from legacy url objects returned by `url.parse()` (which needs to be imported).

Reply:
I don't fully understand what we need here. That's because I'm not an expert of NodeJS, the only thing I know is the [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) class that you can use to parse url string.

Anyway, I will change the name of the local const.

Reply:
> In node.js >= 10.0.0, `URL` is a global with a spec-compliant implementation that does not need to be imported (here, the import could actually be removed). This is different from legacy url objects returned by `url.parse()` (which needs to be imported).

Oh wow, was unaware there was a baked in implementation. That's great to know,

We will still need to import the NodeJS implementation, as once in the cloud we use a stripped down version of node that removes almost all baked in modules expect I think `fs`, and `process`, but we could move our import to import the NodeJS version, so good to know.

@Digitalone1 I won't make you fiddle with this one, since you changed it to something that does at least work. But will test if we can just use Nodes baked in version. Thanks for addressing the change